e2e/test-component/scenarios/embedding-sdk/interactive-question-theming.cy.spec.tsx

```diff
@@ -0,0 +1,140 @@
+          "text-primary": "rgb(255, 255, 255)",
+          brand: "rgb(253, 121, 168)",
+        },
+      };
+
+      cy.get<number>("@questionId").then(questionId => {
+        mountSdkContent(
+          <Box bg={theme.colors?.background} h="100vh">
+            <InteractiveQuestion questionId={questionId} />
+          </Box>,
+          { theme },
+        );
+      });
+
+      getSdkRoot().within(() => {
+        cy.findByText("Product ID").should("be.visible");
+
+        const buttonHoverBg = "rgb(33, 39, 43)";
+        const customColumn = "[aria-label='Custom column']";
+
+        // Should be the lightened version of the background color
+        cy.findByTestId("notebook-button")
+          .should("be.visible")
+          .realHover()
+          .should("have.css", "background-color", buttonHoverBg);
+
+        // Should be the lightened version of the background color
+        cy.findByTestId("chart-type-selector-button")
+          .should("be.visible")
+          .realHover()
+          .should("have.css", "background-color", buttonHoverBg);
+
+        cy.findByTestId("notebook-button").click();
+
+        // Should be the lightened version of the background color, same as the notebook button hover.
+        cy.get(customColumn).should(
+          "have.css",
+          "background-color",
+          buttonHoverBg,
+        );
+
+        // Hover should be a less lightened version of the background color.
+        cy.get(customColumn)
+          .should("be.visible")
+          .realHover()
+          .should("have.css", "background-color", "rgb(31, 36, 41)");
+      });
+    });
+
+    it("derives dynamic css variables for light theme", () => {
+      const theme: MetabaseTheme = {
+        colors: {
+          background: "rgb(255, 255, 255)",
+          "background-hover": "rgb(245, 245, 245)",
+          "background-disabled": "rgb(230, 230, 230)",
+          "text-primary": "rgb(51, 51, 51)",
+          brand: "rgb(253, 121, 168)",
+        },
+      };
+
+      cy.get<number>("@questionId").then(questionId => {
+        mountSdkContent(
+          <Box bg={theme.colors?.background} h="100vh">
+            <InteractiveQuestion questionId={questionId} />
+          </Box>,
+          { theme },
+        );
+      });
+
+      getSdkRoot().within(() => {
+        cy.findByText("Product ID").should("be.visible");
+
+        const customColumn = "[aria-label='Custom column']";
+
+        cy.findByTestId("notebook-button").click();
+
+        // Should be the slightly darker version of the background color, same as the notebook button hover
+        cy.get(customColumn).should(
+          "have.css",
+          "background-color",
+          "rgb(242, 242, 242)",
+        );
+
+        // Hover should be an even darker version of the background color
+        cy.get(customColumn)
+          .should("be.visible")
+          .realHover()
+          .should("have.css", "background-color", "rgb(230, 230, 230)");
```

Comment:
Only the custom column colors are defined as dynamic CSS variables for dark mode, so we only assert those two.